ui/apps/platform/src/Containers/AccessControl/AuthProviders/AuthProviderForm.tsx

```diff
@@ -483,53 +486,123 @@ function AuthProviderForm({
+                                    returned from the authentication provider."
                             >
-                                {selectedAuthProvider.requiredAttributes.map(
-                                    (attribute, index: number) => (
-                                        <Flex
-                                            key={`${attribute.attributeKey}_required_attribute_${index}`}
-                                        >
-                                            <FormGroup label="Key" fieldId={attribute.attributeKey}>
-                                                <TextInput
-                                                    type="text"
-                                                    id={attribute.attributeKey}
-                                                    value={attribute.attributeKey}
-                                                    isDisabled
-                                                />
-                                            </FormGroup>
-                                            <FormGroup
-                                                label="Value"
-                                                fieldId={attribute.attributeValue}
-                                            >
-                                                <TextInput
-                                                    type="text"
-                                                    id={attribute.attributeValue}
-                                                    value={attribute.attributeValue}
-                                                    isDisabled
-                                                />
-                                            </FormGroup>
-                                        </Flex>
-                                    )
+                                <p>
+                                    In case a required attribute is not returned from the
+                                    authentication provider, the login attempt will fail and no role
+                                    will be assigned to the user.
+                                </p>
+                            </Alert>
+                            {(!values.requiredAttributes ||
+                                values.requiredAttributes.length === 0) && (
+                                <p>No required attributes defined</p>
+                            )}
+                            <FieldArray
+                                name="requiredAttributes"
+                                render={(arrayHelpers) => (
+                                    <>
+                                        {values.requiredAttributes &&
+                                            values.requiredAttributes.map(
+                                                (attribute, index: number) => (
+                                                    <Flex key={`required_attribute_${index}`}>
```

Comment:
A general rule about `keys` is that an index shouldn't be used because the position of an element in an array can be changed. Something that was once in index `1` could be moved to index `2`. This negatively impacts performance and causes issue with component state. 

Suggestion: Use `attribute.attributeKey` if it is unique. If not, then we will need to generate a unique `id`

Reply:
@ivan-degtiarenko mentioned that it's not as easy to resolve this without changing the API, and that comes with a bunch of additional problems. For now, we'll leave this as is